provider/cloudflare/cloudflare.go

```diff
@@ -112,8 +111,8 @@ type CloudFlareProvider struct {
 
 // cloudFlareChange differentiates between ChangActions
 type cloudFlareChange struct {
-	Action            string
-	ResourceRecordSet []cloudflare.DNSRecord
+	Action         string
+	ResourceRecord cloudflare.DNSRecord
```

Comment:
This seems like a good change to me. I'm curious why it was an array and if you see any case in which keeping the array would be useful.

Reply:
I think it's because in other providers single record can hold multiple values, and in case of cloudflare single record can hold single value. To have multiple values for single dns name in cloudflare you actually need to create multiple records. So in this case ResourceRecordSet would always have only one value.